src/main/java/me/moodcat/core/mappers/GuiceProvisionExceptionMapper.java

```diff
@@ -15,29 +19,40 @@
  * status code.
  */
 @Provider
+@Slf4j
 public class GuiceProvisionExceptionMapper extends
         AbstractExceptionMapper<ProvisionException> {
 
-    private final NotAuthorizedExceptionMapper notAuthorizedExceptionMapper;
+    private final Map<Class<? extends Exception>, AbstractExceptionMapper<? extends Exception>> mappers;
 
     @Inject
-    public GuiceProvisionExceptionMapper(
-            final NotAuthorizedExceptionMapper notAuthorizedExceptionMapper) {
-        this.notAuthorizedExceptionMapper = notAuthorizedExceptionMapper;
+    public GuiceProvisionExceptionMapper(final NotAuthorizedExceptionMapper notAuthorizedExceptionMapper) {
+        mappers = Maps.newHashMap();
+        
+        mappers.put(NotAuthorizedException.class, notAuthorizedExceptionMapper);
```

Comment:
This could later be changed to some version of observer pattern.
However since this won't be used for quite some time, it would just make it more complicated.